source/fuzz/transformation_access_chain.cpp

```diff
@@ -22,21 +22,82 @@
 namespace spvtools {
 namespace fuzz {
 
+namespace {
+// Given a context, a value and a type id, returns the id of an integer
+// constant with the same type and given value.
+// The type id must correspond to an integer type.
+uint32_t FindIntConstant(opt::IRContext* ir_context, uint32_t value,
+                         uint32_t int_type_id) {
+  auto int_type_inst = ir_context->get_def_use_mgr()->GetDef(int_type_id);
+
+  assert(int_type_inst && "The given type id must exist.");
+
+  auto int_type = ir_context->get_type_mgr()
+                      ->GetType(int_type_inst->result_id())
+                      ->AsInteger();
+
+  assert(int_type && "The given type id must correspond to an integer type.");
+
+  opt::analysis::IntConstant bound_minus_one(int_type, {value});
+
+  // Check that the constant exists in the module
+  if (!ir_context->get_constant_mgr()->FindConstant(&bound_minus_one)) {
+    return 0;
+  }
+
+  return ir_context->get_constant_mgr()
+      ->GetDefiningInstruction(&bound_minus_one)
+      ->result_id();
+}
+
+// TODO: This is copied from TransformationAddFunction. Move it somewhere
+// where it can be accessed by both.
+uint32_t GetBoundForCompositeIndex(
+    opt::IRContext* ir_context, const opt::Instruction& composite_type_inst) {
+  switch (composite_type_inst.opcode()) {
+    case SpvOpTypeArray:
```

Comment:
I think fuzzer_util.cpp would be a good place for this as well.

Reply:
Moved it there, removed it from both TransformationAccessChain and TransformationAddFunction and renamed all the references.